django/db/models/fields/__init__.py

```diff
@@ -2397,9 +2397,12 @@ def as_sql(self, compiler, connection):
             tzname = timezone.get_current_timezone_name() if settings.USE_TZ else None
             sql, tz_params = connection.ops.datetime_extract_sql(self.lookup_name, sql, tzname)
             params.extend(tz_params)
-        else:
-            # DateField and TimeField.
+        elif isinstance(lhs_output_field, DateField):
             sql = connection.ops.date_extract_sql(self.lookup_name, sql)
+        elif isinstance(lhs_output_field, TimeField):
```

Comment:
Would it be better to use the `date_extract_sql()` method for `TimeField` if the `time_extract_sql()` method doesn't exist? If I understand correctly, the backends besides SQLite wouldn't need to be modified. The new method should be mentioned in the "Database backend API" of the 1.9 release notes.